### PR TITLE
refactor(foundation-demo): centralize all footer code 

### PR DESCRIPTION
**Pull request by damienwebdev**
refactor(foundation-demo): centralize all footer code into a single module in foundation-demo/core

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently, the` foundation-demo` has footer related code all over the place. This architecture decision is a side-effect of the way loading styles are managed across components. Instead of pursuing a more complex code structure to mitigate a UX issue, we should try to solve the UX issue directly.

I've opened #66 to discuss this.

## What is the new behavior?
The centralizes all the footer code into the `foundation-demo/core` in both `template` and `footer`.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: apps/foundation-demo/src/app/core/footer/footer.module.spec.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { FooterModule } from './footer.module';
+
+describe('FooterModule', () => {
```
Comment (lderrickable) [inline apps/foundation-demo/src/app/core/footer/footer.module.spec.ts]: I don't think we need this. The build will just fail and the app won't work if the imports are wrong

Comment (damienwebdev) [reply apps/foundation-demo/src/app/core/footer/footer.module.spec.ts]: @lderrickable I'd agree. But I'd also like to understand why it's now auto generated by `ng g m`

**Comment by damienwebdev** [comment]
@lderrickable this needs re-reviewed

**Comment by lderrickable** [comment]
I removed an fdescribe

**Pull request merged**